### ISSUE TITLE
Unable to reply to HTML Formatted emails received from Outlook Mail

### ISSUE THREAD

**Issue by tashecooke** [roundcube/roundcubemail#7494]
Working through the Browser console for errors I found a 403 error for the server response to a Roundcube POST Request.
In looking through Network > Preview /Response for further detail, I have found it to be a Firewall Response( Wordfence ) to a XSS in POST body  - here is the WordPress back end firewall message :

**Wordfence  Traffic Blocking** 

Christchurch, New Zealand
/roundcube/?_task=mail&_unlock…	7/17/2020 7:22:22 PM	122.59.155.125	...125-adsl.sparkbb.co.nz	403	
Type: Blocked
Activity Detail
 Christchurch, New Zealand left https://nzwebxpress.com/roundcube/?_task=mail&_action=compose&_id=5835812945f11519e47581 and was blocked by firewall for XSS: Cross Site Scripting in POST body: _message=%3Cp%3E%26nbsp%3B%3C%2Fp%3E%0D%0A%3Cdiv%20id%3D%22_rc_sig%22%3E%26nbsp%3B%3C%2Fdiv%3E%0D%0A%3Cp%3E%2… at https://nzwebxpress.com/roundcube/?_task=mail&_unlock=loading1594970543024&_framed=1&_lang=en_GB
7/17/2020 7:22:22 PM (56 minutes ago)  
IP: 122.59.155.125 Hostname: 122-59-155-125-adsl.sparkbb.co.nz
Human/Bot: Human
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36

What I am keen to understand is why this happens only to Outlook emails that I am replying to - as previously pointed out.

I would appreciate your insight before I exclude these parameters in Wordfence Filtering .

_Originally posted by @tashecooke in https://github.com/roundcube/roundcubemail/issues/7488#issuecomment-659963467_

**Comment by alecpl** [comment]
We need to find out which part of the request precisselly is causing the problem. I can't do this for you. 

Use "Source code" button in the HTML editor toolbar to modify the HTML content. I propose to cut the code one chunk at a time and click the Save button every time. With this you should be able to narrow it down to the specific line(s).

**Comment by tashecooke** [comment]
I have attached the Form Data for the POST that was  flagged as 403 by Wordfence as indicated above ,
 https://nzwebxpress.com/roundcube/?_task=mail&_unlock=loading1594970543024&_framed=1&_lang=en_GB

**The URL POST Request :**

_token: 8kGSUE1AoM99jHZ9D6QT5RP1i5o16jZu
_task: mail
_action: send
_id: 14750474785f12722f65c24
_attachments: 
_from: 2
_to: geoff.cooke@xtra.co.nz,
_cc: 
_bcc: 
_replyto: 
_followupto: 
_subject: Re: 
_draft_saveid: 
_draft: 
_is_html: 1
_framed: 1
_message: <p>&nbsp;</p>
<div id="_rc_sig">&nbsp;</div>
<p>&nbsp;</p>
<p id="reply-intro">On 2020-07-18 12:47, geoff.cooke@xtra.co.nz wrote:</p>
<blockquote>
<div id="replybody1">
<style type="text/css">@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4}

#replybody1 p.v1MsoNormal, #replybody1 li.v1MsoNormal, #replybody1 div.v1MsoNormal
	{margin:0cm;
	margin-bottom:.0001pt;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;
	mso-fareast-language:EN-US}
#replybody1 span.v1EmailStyle17
	{mso-style-type:personal-compose;
	font-family:"Calibri",sans-serif;
	color:black;
	font-weight:normal;
	font-style:normal}
#replybody1 .v1MsoChpDefault
	{mso-style-type:export-only;
	font-family:"Calibri",sans-serif;
	mso-fareast-language:EN-US}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:72.0pt 72.0pt 72.0pt 72.0pt}
#replybody1 div.v1WordSection1
	{page:WordSection1}</style>
<div lang="EN-GB">
<div class="v1WordSection1">
<p class="v1MsoNormal"><span style="font-size: 12.0pt; color: black;">&nbsp;</span></p>
<p class="v1MsoNormal">&nbsp;</p>
</div>
</div>
</div>
</blockquote>
editorSelector: html
_mdn: 
_dsn: 
_priority: 0
_store_target: Sent

**The Encoded POST Request** 
(as you can see the message body is being picked up by Wordfence as a XSS  - as previously stated )

_token: 8kGSUE1AoM99jHZ9D6QT5RP1i5o16jZu
_task: mail
_action: send
_id: 14750474785f12722f65c24
_attachments: 
_from: 2
_to: geoff.cooke%40xtra.co.nz%2C
_cc: 
_bcc: 
_replyto: 
_followupto: 
_subject: Re%3A+
_draft_saveid: 
_draft: 
_is_html: 1
_framed: 1
_message: %3Cp%3E%26nbsp%3B%3C%2Fp%3E%0D%0A%3Cdiv+id%3D%22_rc_sig%22%3E%26nbsp%3B%3C%2Fdiv%3E%0D%0A%3Cp%3E%26nbsp%3B%3C%2Fp%3E%0D%0A%3Cp+id%3D%22reply-intro%22%3EOn+2020-07-18+12%3A47%2C+geoff.cooke%40xtra.co.nz+wrote%3A%3C%2Fp%3E%0D%0A%3Cblockquote%3E%0D%0A%3Cdiv+id%3D%22replybody1%22%3E%0D%0A%3Cstyle+type%3D%22text%2Fcss%22%3E%40font-face%0D%0A%09%7Bfont-family%3A%22Cambria+Math%22%3B%0D%0A%09panose-1%3A2+4+5+3+5+4+6+3+2+4%7D%0D%0A%40font-face%0D%0A%09%7Bfont-family%3ACalibri%3B%0D%0A%09panose-1%3A2+15+5+2+2+2+4+3+2+4%7D%0D%0A%0D%0A%23replybody1+p.v1MsoNormal%2C+%23replybody1+li.v1MsoNormal%2C+%23replybody1+div.v1MsoNormal%0D%0A%09%7Bmargin%3A0cm%3B%0D%0A%09margin-bottom%3A.0001pt%3B%0D%0A%09font-size%3A11.0pt%3B%0D%0A%09font-family%3A%22Calibri%22%2Csans-serif%3B%0D%0A%09mso-fareast-language%3AEN-US%7D%0D%0A%23replybody1+span.v1EmailStyle17%0D%0A%09%7Bmso-style-type%3Apersonal-compose%3B%0D%0A%09font-family%3A%22Calibri%22%2Csans-serif%3B%0D%0A%09color%3Ablack%3B%0D%0A%09font-weight%3Anormal%3B%0D%0A%09font-style%3Anormal%7D%0D%0A%23replybody1+.v1MsoChpDefault%0D%0A%09%7Bmso-style-type%3Aexport-only%3B%0D%0A%09font-family%3A%22Calibri%22%2Csans-serif%3B%0D%0A%09mso-fareast-language%3AEN-US%7D%0D%0A%40page+WordSection1%0D%0A%09%7Bsize%3A612.0pt+792.0pt%3B%0D%0A%09margin%3A72.0pt+72.0pt+72.0pt+72.0pt%7D%0D%0A%23replybody1+div.v1WordSection1%0D%0A%09%7Bpage%3AWordSection1%7D%3C%2Fstyle%3E%0D%0A%3Cdiv+lang%3D%22EN-GB%22%3E%0D%0A%3Cdiv+class%3D%22v1WordSection1%22%3E%0D%0A%3Cp+class%3D%22v1MsoNormal%22%3E%3Cspan+style%3D%22font-size%3A+12.0pt%3B+color%3A+black%3B%22%3E%26nbsp%3B%3C%2Fspan%3E%3C%2Fp%3E%0D%0A%3Cp+class%3D%22v1MsoNormal%22%3E%26nbsp%3B%3C%2Fp%3E%0D%0A%3C%2Fdiv%3E%0D%0A%3C%2Fdiv%3E%0D%0A%3C%2Fdiv%3E%0D%0A%3C%2Fblockquote%3E
editorSelector: html
_mdn: 
_dsn: 
_priority: 0
_store_target: Sent

**Comment by alecpl** [comment]
BTW, I don't see how this code could cause any XSS. All of this looks like a wordfance false positive.

**Comment by tashecooke** [comment]
Thanks, I am wondering about a WordFence false positive.
I will try without any body text ( including auto generated text that states > On 2020-07-18 12:47, geoff.cooke@xtra.co.nz wrote: ). I guess there will still be an encoded  body message.

**Comment by alecpl** [comment]
Most likely it is the `<style>` tag content as this is the part that comes from Outlook.

**Comment by tashecooke** [comment]
What I have found :
I have sent an email from Outlook which has no body text nor subject matter. 
In Roundcube I replied to this email as follows:

- Removed all auto generated text i.e. no body text nor subject Re: - non issues sending a reply.
- Left Roundcube auto generated reply text - error 403 , POST body encoded message seen as XSS by WorFence.

I am out of my depth using Chrome Dev Tools to ,carve up HTML and resubmit - so will need to find a good resource on this before trying to narrow down what part of the HTML is the issue , as you say <style> tag comes from Outlook.
If in fact the style tag is the issue, what would be the next step to resolve this -doe sthis mean I have to set each customers Roundcube settings to never compose reply emails in HTML ?

**Comment by alecpl** [comment]
If we know preciselly which part of the style is the problem we would consider if it is possible to automatically remove that part. Otherwise I don't see what we could do. Removing the whole style tag is a No.

You don't have to use dev tools. "Source code" in the editor toolbar should be easier.

**Comment by tashecooke** [comment]
Sorry I feel I am wasting your time here - I need to bring myself up to speed on Using the  "Source code" button in the HTML editor toolbar to modify the HTML content. I though you where referring to Dev Tools "Sources".
I gather you are referring to some another HTML editor - which I am struggling to identify - any further guidance on that would be appreciated . Are you able to use this editor to resend the modified email HTML to the mail server ?

**Comment by tashecooke** [comment]
Ok, I have figured what you have been trying to say . I can edit the source code in the Roundcube email Editor .

I have remove the following :
<style type="text/css">@font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4}

#replybody1 p.v1MsoNormal, #replybody1 li.v1MsoNormal, #replybody1 div.v1MsoNormal
	{margin:0cm;
	margin-bottom:.0001pt;
	font-size:11.0pt;
	font-family:"Calibri",sans-serif;
	mso-fareast-language:EN-US}
#replybody1 span.v1EmailStyle17
	{mso-style-type:personal-compose;
	font-family:"Calibri",sans-serif;
	color:black;
	font-weight:normal;
	font-style:normal}
#replybody1 .v1MsoChpDefault
	{mso-style-type:export-only;
	font-family:"Calibri",sans-serif;
	mso-fareast-language:EN-US}
@page WordSection1
	{size:612.0pt 792.0pt;
	margin:72.0pt 72.0pt 72.0pt 72.0pt}
#replybody1 div.v1WordSection1
	{page:WordSection1}</style>

This resolves the 403 issue, Removing the whole style tag does not alter other text styling I have made to the text added in the reply email.

**Comment by alecpl** [comment]
And I said that removing the whole style tag is not a solution that we could accept. You have to figure out which part of that content causes the issue.

**Comment by tashecooke** [comment]
I have been through each component between the <style type="text/css"></style>  tags removing and then sending email  until there was nothing between the tags -  it will only send without a 403 when the you finally remove the <style type="text/css">
</style> tags.

**Comment by alecpl** [comment]
Then this looks to me as a wontfix.

**Comment by tashecooke** [comment]
Sorry what do you mean by that ?

**Comment by alecpl** [comment]
I mean that we'll not remove any style tags defined in the HTML content to workaround obvious bug in external software.

**Comment by tashecooke** [comment]
Thanks -  so you see this as an Outlook issue, and I take it that this has not been experienced by others users of Roundcube?

**Comment by alecpl** [comment]
No, this is a Wordfence issue.

**Comment by tashecooke** [comment]
Thanks I will now discuss this with them.
